src/core/thread/mle.cpp

```diff
@@ -3298,20 +3298,19 @@ otError Mle::InformPreviousParent(void)
     messageInfo.SetInterfaceId(netif.GetInterfaceId());
 
     SuccessOrExit(error = netif.GetIp6().SendDatagram(*message, messageInfo, Ip6::kProtoNone));
-    message = NULL;
 
```

Comment:
I don't follow why we need this change...

In previous version, 
- if `SendDatagram()` fails, we would exit immediately (from `SuccessOrExit`) and `message` would stay non-NULL, so it would be freed at `exit` label,
- if `SendDatagram()` is successful, then next line we would set `message = NULL` which would ensure message is not freed. 

Is there something I am missing here?

Reply:
Ah, yes, you're right.  It just does not follow the convention used elsewhere.

FWIW, this eliminates the need to set `message = NULL` and the NULL-check below on the success path.

Reply:
I am ok with this change to follow convention...

The slight benefit I see with the check for message being non-NULL outside of `if (error != OT_ERROR_NONE)` is that exit paths in code can choose to return `OT_ERROR_NONE` as error status while we may still need to free the message (but this should not be common)...